packages/checkbox/src/Checkbox/Checkbox.tsx

```diff
@@ -35,148 +35,158 @@ import { CheckboxProps } from './Checkbox.types';
+      e: React.MouseEvent<HTMLInputElement> & { target: HTMLInputElement },
+    ) => {
+      // Don't fire events if disabled
+      if (disabled) {
+        return;
+      }
+
+      if (onClickProp) {
+        onClickProp(e);
+      }
+
+      // For Microsoft Edge and IE, when checkbox is indeterminate, change event does not fire when clicked.
+      // Explicitly call onChange for this case
+      if (indeterminateProp) {
+        onChange(e);
+        e.stopPropagation();
+      }
+    };
+
+    return (
+      <LeafyGreenProvider
+        baseFontSize={baseFontSize === 16 ? baseFontSize : 14}
+        darkMode={darkMode}
       >
-        <Label
-          id={labelId}
-          htmlFor={checkboxId}
-          disabled={disabled}
-          className={cx(labelStyle, labelHoverStyle[theme], {
-            [disabledLabelStyle]: disabled,
-          })}
+        <div
+          className={cx(
+            containerStyle,
+            {
+              [disabledContainerStyle]: disabled,
+            },
+            className,
+          )}
+          data-lgid={dataLgId}
+          style={style}
         >
-          <input
-            {...rest}
-            id={checkboxId}
-            className={cx(inputClassName, inputStyle, inputFocusStyles[theme])}
-            type="checkbox"
-            name={name}
-            checked={isChecked}
-            aria-label={ariaLabel}
-            aria-disabled={disabled}
-            aria-checked={indeterminateProp ? 'mixed' : isChecked}
-            aria-labelledby={labelId}
-            onClick={onClick}
-            onChange={onChange}
-          />
-
-          <Check
-            theme={theme}
-            isChecked={isChecked}
-            indeterminate={indeterminateProp}
+          <Label
+            id={labelId}
+            htmlFor={checkboxId}
             disabled={disabled}
-            animate={animate}
-            selector={checkWrapperClassName}
-          />
-
-          {label && (
-            <span
-              className={cx(labelTextStyle, {
-                [css`
-                  font-weight: ${fontWeights.regular};
-                `]: !bold,
-              })}
-            >
-              {label}
-            </span>
+            className={cx(labelStyle, labelHoverStyle[theme], {
+              [disabledLabelStyle]: disabled,
+            })}
+          >
+            <input
+              {...rest}
+              id={checkboxId}
+              className={cx(
+                inputClassName,
+                inputStyle,
+                inputFocusStyles[theme],
+              )}
+              type="checkbox"
+              name={name}
+              checked={isChecked}
+              aria-label={ariaLabel}
+              aria-disabled={disabled}
+              aria-checked={indeterminateProp ? 'mixed' : isChecked}
+              aria-labelledby={labelId}
+              onClick={onClick}
+              onChange={onChange}
+              ref={forwardRef}
```

Comment:
Can we add a test that it accepts a ref?

Reply:
Do we usually test that?

Reply:
Only a few. I did it for all table components. However, i don't feel strongly about this.